app/models/post.rb

```diff
@@ -0,0 +1,16 @@
+class Post < ApplicationRecord
+  belongs_to :author, class_name: 'User'
+  has_many :comments
+  has_many :likes
+
+  after_save :update_users_posts_counter
+  after_destroy :update_users_posts_counter
+
+  def recent_comments
+    comments.last(5)
+  end
+
+  def update_users_posts_counter
+    author.update(posts_counter: author.posts.count)
+  end
```

Comment:
- Hey, this method should be a private method since you're calling the `after_save` to update the `update_users_posts_counter` every time a new post has been saved to the database. add the private to the method.

```ruby

private

 def update_users_posts_counter
    author.update(posts_counter: author.posts.count)
  end
```